tool/healthz/main.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2017 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gravitational/trace"
+	minio "github.com/minio/minio-go"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	defaultEndpoint       = "localhost:18080"
+	defaultBucket         = "liveness-check"
+	defaultBucketLocation = "G1"
+	defaultPrefix         = "liveness"
+)
+
+// s3Config describes configuration to access an s3 bucket
+type s3Config struct {
+	bucket string
+	client *minio.Client
+}
+
+func main() {
+	s3AccessKeyID := flag.String("access-key-id", "", "S3 access key")
+	s3SecretAccessKey := flag.String("secret-access-key", "", "S3 secret key")
+	s3Endpoint := flag.String("endpoint", defaultEndpoint, "S3 endpoint address")
+	s3Bucket := flag.String("bucket", defaultBucket, "S3 Bucket name")
+
+	flag.Parse()
+	if *s3AccessKeyID == "" && *s3SecretAccessKey == "" {
+		log.Fatal("access-key-id and secret-access-key are required")
+	}
+
+	client, err := initClient(*s3Endpoint, *s3AccessKeyID, *s3SecretAccessKey)
+	if err != nil {
+		log.Fatal("failed to create s3 client")
```

Comment:
I'd add the error (message) as well - otherwise it's going to be complicated to find out:
```go
	log.Fatalf("failed to create s3 client: %v", err)
```